### PR TITLE
mariadb support + new scripts folder for backup

### PR DESCRIPTION
**Pull request by abhishek-goyal1**

### CODE REVIEW

File: VMBackup/main/workloadPatch/WorkloadPatch.py [hunk 1]
```diff
@@ -54,7 +54,9 @@ def __init__(self, logger):
         self.logbackup = ""
         self.custom_scripts_enabled = 0
         self.scriptpath= "DefaultScripts"
+        self.script_folder= "/etc/azure"
         self.confParser()
+        self.updateScriptsFolder()
```
Comment (soumyapattnaik) [inline VMBackup/main/workloadPatch/WorkloadPatch.py:59]: move it to pre

File: VMBackup/main/workloadPatch/WorkloadPatch.py [hunk 2]
```diff
@@ -381,6 +387,36 @@ def confParser(self):
...
+        copyProcess.wait()
+        changeOwnerProcess = subprocess.Popen(['chown','-R',self.linux_user,newScriptsFolder], stdout=subprocess.PIPE)
+        changeOwnerProcess.wait()
+        permissionProcess = subprocess.Popen(['chmod','-R','700',newScriptsFolder], stdout=subprocess.PIPE)
+        permissionProcess.wait()
+        self.logger.log("WorkloadPatch: Script files copied to temporary scripts folder present at " + newScriptsFolder)
+        
+    
+    def removeScriptsFolder(self):
+        self.logger.log("WorkloadPatch: Removing temporary scripts folder")
```
Comment (soumyapattnaik) [inline VMBackup/main/workloadPatch/WorkloadPatch.py:413]: put a try catch here

File: VMBackup/main/workloadPatch/WorkloadPatch.py [hunk 3]
```diff
@@ -120,19 +126,19 @@ def preMaster(self):
...
             return None
         
-        if 'mysql' in self.name.lower():
+        if 'mysql' in self.name.lower() or 'mariadb' in self.name.lower():
             self.logger.log("WorkloadPatch: Create connection string for premaster mysql")
             if self.outfile == "":
                 self.error_details.append(ErrorDetail(CommonVariables.FailedWorkloadIPCDirectoryMissing, "IPC directory missing"))
                 return None
-            prescript = os.path.join(os.getcwd(), "main/workloadPatch/"+self.scriptpath+"/preMysqlMaster.sql")
+            prescript = os.path.join(self.script_folder, self.scriptpath + "/preMysqlMaster.sql")
```
Comment (soumyapattnaik) [inline VMBackup/main/workloadPatch/WorkloadPatch.py:134]: change it to temp_script_path

File: VMBackup/main/workloadPatch/WorkloadPatch.py [hunk 4]
```diff
@@ -381,6 +387,36 @@ def confParser(self):
             if(self.name != None):
                 self.error_details.append(ErrorDetail(CommonVariables.FailedWorkloadConfParsingError, "exception in workloadconfig parsing"))
     
+    def updateScriptsFolder(self):
```
Comment (soumyapattnaik) [inline VMBackup/main/workloadPatch/WorkloadPatch.py:390]: create temp script folder

File: VMBackup/main/workloadPatch/WorkloadPatch.py [hunk 5]
```diff
@@ -381,6 +387,36 @@ def confParser(self):
...
+            
+        if (os.path.exists(newScriptsFolder)):
+            self.logger.log("WorkloadPatch: Existing temporary scripts folder found..removing")
+            self.removeScriptsFolder()
+            
+        copyProcess = subprocess.Popen(['cp','-ar',originalScriptsFolder,self.script_folder])
+        copyProcess.wait()
+        changeOwnerProcess = subprocess.Popen(['chown','-R',self.linux_user,newScriptsFolder], stdout=subprocess.PIPE)
+        changeOwnerProcess.wait()
+        permissionProcess = subprocess.Popen(['chmod','-R','700',newScriptsFolder], stdout=subprocess.PIPE)
```
Comment (soumyapattnaik) [inline VMBackup/main/workloadPatch/WorkloadPatch.py:407]: check for read and execute permission only

File: VMBackup/main/workloadPatch/WorkloadPatch.py [hunk 6]
```diff
@@ -381,6 +387,36 @@ def confParser(self):
             if(self.name != None):
                 self.error_details.append(ErrorDetail(CommonVariables.FailedWorkloadConfParsingError, "exception in workloadconfig parsing"))
     
+    def updateScriptsFolder(self):
+        self.logger.log("WorkloadPatch: Creating temporary scripts folder")
```
Comment (soumyapattnaik) [inline VMBackup/main/workloadPatch/WorkloadPatch.py:391]: try-catch handling

File: VMBackup/main/workloadPatch/WorkloadPatch.py (hunk 5 above)
Comment (abhishek-goyal1) [reply VMBackup/main/workloadPatch/WorkloadPatch.py]: Works with only read and execute, modified as per.

**Review by soumyapattnaik — APPROVED** [review]

**Pull request merged**